bindings/matlab/+iDynTreeWrappers/getMeshes.m

```diff
@@ -1,118 +1,147 @@
+    % get the linkSolidShapes containing the mesh information
+    visual           = model.visualSolidShapes;
+    linkSolidShapesV = visual.linkSolidShapes;
+    
+    % get number of links
+    numberOfLinks    = linkSolidShapesV.size;
+    linkMeshInfo     = struct('meshInfo',{},'linkName',{});
+    
+    % get pointer to beggining of vector
+    iterator         = linkSolidShapesV.begin;
+    
+    % iterate getting the name of the mesh and the name of the link
+    count               = 1;
+    link_with_no_visual = [];
+    
+    for links = 1:numberOfLinks
+        
+        linkName                     = model.getLinkName(links-1);
+        linkMeshInfo(links).linkName = linkName;
+        solidarray                   = iterator.next;
+        solids_number                = size(solidarray,2);
+        meshInfo                     = struct('meshFile',{},'mesh_triangles',{},'link_H_geom',{});
+        
+        for solids = 1:solids_number
+            
+            if solidarray{solids}.isExternalMesh
+                
+                externalMesh = solidarray{solids}.asExternalMesh;
+                scale        = externalMesh.scale.toMatlab;
+                meshName     = split(externalMesh.filename,':');
+                meshFile     = meshName{2};
+                
+                % Import an STL mesh, returning a PATCH-compatible face-vertex structure
+                if strcmp('package',meshName{1})
+                    
+                    mesh_triangles = stlread([meshFilePrefix meshFile]);
+                else
+                    mesh_triangles = stlread(meshFile);
+                end
+                
+                meshInfo(solids).meshFile = meshFile;
+                meshInfo(solids).scale    = scale';
             else
-                mesh_triangles = stlread(meshFile);
-            end
-            meshInfo(solids).meshFile=meshFile;
-            meshInfo(solids).scale=scale';
-        else
-            meshInfo(solids).scale=[1,1,1];
-            if solidarray{solids}.isCylinder
-                meshInfo(solids).meshFile='cylinder';
-                length=solidarray{solids}.asCylinder.length;
-                radius=solidarray{solids}.asCylinder.radius;
-                mesh_triangles=calculateMeshFromCylinder(length,radius);
-            end
-            if solidarray{solids}.isBox
-                meshInfo(solids).meshFile='box';
-                box_dimensions(1)=solidarray{solids}.asBox.x;
-                box_dimensions(2)=solidarray{solids}.asBox.y;
-                box_dimensions(3)=solidarray{solids}.asBox.z;
-                mesh_triangles=calculateMeshFromBox(box_dimensions);
-            end
-            if solidarray{solids}.isSphere
-                meshInfo(solids).meshFile='sphere';
-                radius=solidarray{solids}.asSphere.radius;
-                mesh_triangles=calculateMeshFromSphere(radius);
+                meshInfo(solids).scale    = [1,1,1];
+                
```

Comment:
Nitpick: Please avoid commiting trailing whitespaces, as they can create problems in comparing code, creating spurious whitespace only changes that to not correspond to real changes (see https://github.com/robotology/idyntree/pull/744/files#diff-eddfdc7bb22a31340c851c77734e51c6L27). Most editor have options to automatically avoid to commit trailing whitespaces, but after a quick search I am afraid that MATLAB does not.